File: backend/AchievementOffice/Entities/AchievementVerificationRequest.cs
namespace AchievementOffice.Entities;

public class AchievementVerificationRequest
{
    public Guid Id { get; set; }

    public Guid AchievementId { get; set; }
    public Achievement Achievement { get; set; } = null!;

    public Guid RequesterUserId { get; set; }
    public User RequesterUser { get; set; } = null!;

    public Guid TargetUserId { get; set; }
    public User TargetUser { get; set; } = null!;

    public VerificationRequestStatus Status { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime? RespondedAt { get; set; }
}

public enum VerificationRequestStatus
{
    Pending = 0,
    Approved = 1,
    Denied = 2
}

File: frontend/src/types/achievementVerificationRequest.ts
export interface AchievementVerificationRequest {
    id: string;
    achievementId: string;
    achievementTitle: string;
    requesterUserId: string;
    requesterLogin: string;
    targetUserId: string;
    targetUserLogin: string;
    status: string;
    createdAt: string;
    respondedAt?: string | null;
}

export interface UserReviewer {
    userId: string;
    login: string;
    firstName: string;
    lastName: string;
}